components-api/src/main/scala/pl/touk/nussknacker/engine/api/component/ProcessingMode.scala

```diff
@@ -32,4 +34,13 @@ object ProcessingMode {
     case other                               => throw new IllegalArgumentException(s"Not known processing mode: $other")
   }
 
+  sealed trait AllowedProcessingModes
```

Comment:
I think that this sealed trait hierarchy is more related to `Component` than to `ProcessingMode` (even it uses the `ProcessingMode` case class). WDYT about moving it close to the `Component` definition?